### PR TITLE
CI: Use shared versions GH Actions workflow

### PR DESCRIPTION
**Pull request by olleolleolle**
fix ruby/strscan#73

I took some inspiration from https://github.com/ruby/uri/blob/master/.github/workflows/test.yml and then I edited the necessary values.

(I put this as a Draft PR, since I am not sure about all the details, which engines and versions will be included.)

### CODE REVIEW

**Comment by eregon** [comment]
FYI, CI fails on jruby latest release, tracked as https://github.com/ruby/strscan/issues/75

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,35 +5,28 @@ on:
...
 jobs:
+  ruby-versions:
+    uses: ruby/actions/.github/workflows/ruby_versions.yml@master
+    with:
+      min_version: 2.4
+
   host:
-    name: ${{ matrix.os }} ${{ matrix.ruby }}
+    needs: ruby-versions
+    name: build (${{ matrix.ruby }} / ${{ matrix.os }})
```
Comment (kou) [inline .github/workflows/ci.yml:15]: Can we keep the original `name:` for short label?

**Review by kou — COMMENTED** [review]
@headius Do we need both of `jruby` and `jruby-head`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (olleolleolle) [reply .github/workflows/ci.yml]: Yes, now done!

**Comment by olleolleolle** [comment]
I learned about the JRuby detail:

> Yeah we had a transition period where it supported both old and new but now all released JRuby should be on **the new version**.

And by "the new version" headius is referring to the Joni library version.

**Comment by headius** [comment]
I have merged #76 and master is now green on latest JRuby.

**Comment by headius** [comment]
@olleolleolle Thanks for the PR!

**Comment by olleolleolle** [comment]
In #82, kou explained the reason for the failed builds on current master (and also in this PR).

I will merge this, now.

**Pull request merged**